reactor-pool/src/test/java/reactor/pool/CommonPoolTest.java

```diff
@@ -1117,6 +1119,31 @@ void discardCloseableWhenCloseFailureLogs(PoolStyle configAdjuster) {
 		}
 	}
 
+	@ParameterizedTestWithName
+	@MethodSource("allPools")
+	void pendingTimeout(PoolStyle configAdjuster) throws Exception {
+		PoolBuilder<String, ?> builder = PoolBuilder
+				.from(Mono.just("pendingTimeout"))
+				.sizeBetween(0, 1)
+				.maxPendingAcquire(10);
+		AbstractPool<String> pool = configAdjuster.apply(builder);
+
+		CountDownLatch latch = new CountDownLatch(3);
+		ExecutorService executorService = Executors.newFixedThreadPool(20);
+		CompletableFuture<?>[] completableFutures = new CompletableFuture<?>[4];
+		for (int i = 0; i < completableFutures.length; i++) {
+			completableFutures[i] = CompletableFuture.runAsync(
+					() -> pool.acquire(Duration.ofMillis(10))
+							.doOnError(t -> latch.countDown())
+							.onErrorResume(PoolAcquireTimeoutException.class, t -> Mono.empty())
+							.block(),
+					executorService);
+		}
+
+		Mono.fromCompletionStage(CompletableFuture.allOf(completableFutures)).block(Duration.ofSeconds(5));
```

Comment:
The mix of `CompletionStage`/`CompletableFuture` APIs makes the test a bit harder to follow. Consider picking a single API for the infrastructure around the actual SUT. With reactor the parallelization can be also achieved with a `flatMap`, while with `CompletableFuture.allOf()` you can call `join()` on them.

Reply:
used `join()`